### DISCUSSION TITLE
Current Date/Time [ANSWERED]

### DISCUSSION THREAD

**Discussion by vinyli85** [pi3d/pi3d_demos#36] [Q&A] [answered]
Hey,

is it possible to have the current Date/Time displayed at the top or bottom?
Maybe you can point me where to put the code. I tried to find how to to that (goggled a bit) but did not find anything that worked. My programming skills are almost zero :)

**Comment by paddywwoof** [comment]
Yes it wouldn't be too hard to put the time on. It's basically the same code as for the name/date/location but a bit simpler, without fading and probably with a separate thread to refresh it every second. I'll post something here when I get a moment (probably not today now!)

**Comment by paddywwoof** [comment, marked as answer]
Hi, this is the kind of thing you could do
```python
import threading #this should be at the top with other imports
...
text_bkg.set_shader(back_shader)
text_bkg.set_material((0, 0, 0))
################################################################################
# show time code ###############################################################
tm_text = pi3d.PointText(font, CAMERA, max_chars=10, point_size=config.SHOW_TEXT_SZ)
tm_textblock = pi3d.TextBlock(x=-DISPLAY.width * 0.5 + 50, y=DISPLAY.height * 0.4,
                          z=0.1, rot=0.0, char_count=9,
                          text_format="{}".format(" "), size=0.99,
                          spacing="F", space=0.02, colour=(1.0, 1.0, 1.0, 1.0))
tm_text.add_text_block(tm_textblock)
tm_text_bkg = pi3d.Sprite(w=120, h=40, x=-DISPLAY.width * 0.5 + 100, y=DISPLAY.height * 0.4, z=4.0)
tm_text_bkg.set_shader(back_shader)
tm_text_bkg.set_material((0, 0, 0, 0.3))

def update_tm_text():
  while True:
    time.sleep(5.0) # only showing HH.MM so accurate to 5s OK
    tm = time.localtime()
    tm_textblock.set_text(text_format="{:02d}.{:02d}".format(tm.tm_hour, tm.tm_min)) # or put colon in CODEPOINTS
    tm_text.regen()
    if quit:
      break

tm_thread = threading.Thread(target=update_tm_text)
tm_thread.start()
################################################################################

num_run_through = 0
while DISPLAY.loop_running():
...
    if len(textblock.text_format.strip()) > 0: #only draw background if text there
      text_bkg.draw()

  text.draw()
  ##############################################################################
  tm_text_bkg.draw()
  tm_text.draw()
  ##############################################################################

  if config.KEYBOARD:
    k = kbd.read()

```
So you can see I basically just copied the text, textblock and text_bkg section and changed the location. x,y = (0,0) is in the middle of the screen with x increasing to right and y increasing upwards The threading function breaks when the variable ```quit``` is set to True by an MQTT message, for testing you might want to set quit true if the escape key is pressed.
```
    if k==27: #ESC
      quit = True
      break
```

**Reply by vinyli85** [reply]
Thanks, that's what I was looking for. It is working so far, only thing I have "trouble" with is to have a colon as a separator. I tried following but it doesn't work as expected:
~~`tm_textblock.set_text(text_format="{:02d}'\u003A'{:02d}".format(tm.tm_hour, tm.tm_min))`~~
~~As I understand `'\u003A'` should show a colon, right?~~

**Comment by paddywwoof** [comment]
Look at the bottom of PictureFrame2020config.py those CODEPOINTS are the characters available to use. If you want to use a colon just add it to the string! (then just ```"{:02d}:{:02d}"``` will work)

Paddy
PS in python3 you can put unicode into a string using the \u prefix so ```"hello\u003Aworld"``` results in ```hello:world```. Because they're inside ```"``` quotes your apostrophes will just get included in the string ```hello':'world```. Also, with PointText and TextBlock you have to specify the number of characters available when they are created; in the example code 10 and 9 so extra characters ('AM' or 'GMT' etc) might require a bigger capacity to be specified initially.

**Reply by vinyli85** [reply]
Thank again. 
I played a bit with the code and I found a position and size for the time to show.

PS
Wanted to mark your comment as answer but it looks like this function is not activated

**Comment by paddywwoof** [comment]
No, I can't see where to enable that. I think ```Discussion``` is fairly beta at the moment. Also I set it to be general chit-chat mode rather than stack overflow style question and answer with votes - which was the other option. I might try switching it over to see what it looks like. In the mean time I've edited your title!
Paddy

**Comment by paddywwoof** [comment]
OK the alternative category is Q&A and that is set up in the mode for ticking answers already. I've moved these two threads from Ideas to Q&A
